src/typetree.jl

```diff
@@ -63,70 +63,86 @@ function to_md(tt::TypeTree, ctx)
     return LLVM.Metadata(LLVM.MetadataAsValue(ccall((:EnzymeTypeTreeToMD, API.libEnzyme), LLVM.API.LLVMValueRef, (API.CTypeTreeRef,LLVM.API.LLVMContextRef), tt, ctx)))
 end
 
-function typetree(::Type{T}, ctx, dl, seen=nothing) where T <: Integer
+const TypeTreeTable = IdDict{Any, Union{Nothing, TypeTree}}
```

Comment:
Minor but could we have this be Type as the key?

Reply:
Doesn't really matter since we use objectid